### ISSUE TITLE
B008 false positive in FastAPI / Typer

### ISSUE THREAD

**Issue by johnthagen** [PyCQA/flake8-bugbear#215]
In FastAPI it is very common to use class objects as special, succicient indicators to the framework.

- https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#use-query-as-the-default-value

```py
@app.get("/items/")
async def read_items(q: Optional[str] = Query(None, max_length=50)):
```

Typer exposes the same, convenient convenient user experience:

- https://typer.tiangolo.com/tutorial/arguments/optional/#an-alternative-cli-argument-declaration

```py
def main(name: str = typer.Argument(...)):
    typer.echo(f"Hello {name}")


if __name__ == "__main__":
    typer.run(main)
```

`flake8-bugbear` will throw `B008` errors at this usage. It would be nice if `flake8-bugbear` whitelisted these or allowed the user to whitelist certain types.

**Comment by johnthagen** [comment]
I now see that `extend-immutable-calls` exists. I think the docs could use an example for how this could be used (if possible?) on third-party types.

I would be happy to submit a PR if someone can give me a lead on how this configuration could be used with FastAPI or Typer types.

**Comment by johnthagen** [comment]
```ini
[flake8]
...
extend-immutable-calls =
    Argument
```

This worked for the `typer` example. I will submit a PR if a maintainer can confirm that this is intended/correct usage for the problem in this issue.

**Comment by npip99** [comment]
If anyone is using `pyproject.toml`,

```toml
[tool.ruff.lint.flake8-bugbear]
extend-immutable-calls = [
    # Whitelisted default arguments
    "fastapi.Header",
    "fastapi.Depends",
]
```
